### PR TITLE
Feature/cordova plugin

### PR DESCRIPTION
**Pull request by dominikfeininger**

### CODE REVIEW

**Comment by vobu** [comment]
fat fingers, sorry :(

File: lib/cordova-plugins/factory.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
+const logger = require('../Logger');
+
+/**
+ * plugin mock factory
+ * to create a proper abstraction for the test, the test does not neet to know whrere it's mocks come from !
```
Comment (vobu) [inline lib/cordova-plugins/factory.js:5]: "the test doesn't need to know where the mock happens at runtime"

File: lib/cordova-plugins/factory.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+ *
+ * * How to mock a plugin
+ * ----------------------
+ * 1. Write its name, need to match the cordova plugin name and the actual mock filename, into the `wdi5: plugins: <pluginname>` config.
```
Comment (vobu) [inline lib/cordova-plugins/factory.js:9]: put the exact cordova plugin name and its' mock countpart into: wdi5: { plugins: { <pluginName>: {} } }

File: lib/cordova-plugins/factory.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+ * 1.1 This triggers the load of the javascript file
```
Comment (vobu) [inline lib/cordova-plugins/factory.js:10]: -> maps source to mock at wdi5 runtime

File: lib/cordova-plugins/factory.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        electron: {}
+    },
+    // index
+    _: null,
+    _context: null,
+    // name of platform under test
+    _platform: '',
+
+    /**
+     * call to trigger the init process
```
Comment (vobu) [inline lib/cordova-plugins/factory.js:35]: missing param description for _, missing param for context

File: lib/cordova-plugins/factory.js [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+        if (pluginConfigList) {
+            Object.keys(pluginConfigList).forEach((name) => {
+                // load plugin
+                this._loadPluginWithName(name);
+            });
+
+            return pluginConfigList;
+        } else {
+            // no plugin config found
+            logger.log('no plugins in config defined');
```
Comment (vobu) [inline lib/cordova-plugins/factory.js:131]: eventually return []?

File: lib/cordova-plugins/factory.js [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+                // create cordova plugin mock as base
+                if (!window.cordova) {
+                    window.cordova = {
+                        plugins: {}
+                    };
+                }
+
+                try {
+                    // attach
+                    const result = eval(sSetupFunction)();
```
Comment (vobu) [inline lib/cordova-plugins/factory.js:163]: unnecessary variable declaration -> result is never used

File: test/ui5-app/webapp/test/e2e/test-plugin.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
+const assert = require('assert');
+const wdi5 = require('../../../../../index');
+
+describe('ui5 plugin', () => {
+    const viewName = 'test.Sample.view.Main';
+
+    it('test the cordova barcodescanner plugin', () => {
```
Comment (vobu) [inline test/ui5-app/webapp/test/e2e/test-plugin.js:7]: "test the _mocked_ cordova barcodescanner plugin"?

File: test/wdio-shared.conf.js [hunk 8]
```diff
@@ -97,10 +97,19 @@ exports.config = {
     wdi5: {
         screenshotPath: path.join('test', 'report', 'screenshots'),
         deviceType: 'native',
-        logLevel: 'error',
+        logLevel: 'verbose',
         capabilities: {
             rotate: true,
             camera: 2
+        },
+        plugins: {
```
Comment (vobu) [inline test/wdio-shared.conf.js:105]: the plugins section is both in wdio-shared.conf.js + wdio.conf.js - wouldn't it be sufficient in only the shared config?

File: index.js [hunk 9]
```diff
@@ -70,19 +75,27 @@ class _ {
...
+module.exports = (context, webcontext) => {
     if (!_instance) {
         // create new if parameters are supplied
         _instance = new _(webcontext);
+
+        // UI5 bridge setup
+        _instance.getWDioUi5().setup(context); // use wdio hooks for setting up wdio<->ui5 bridge
+        _instance.getWDioUi5().injectUI5(context); // needed to let the instance know that UI5 is now available for work
+
+        // create plugins
```
Comment (vobu) [inline index.js:87]: create _mocks_ for plugins

File: index.js [hunk 10]
```diff
@@ -1,43 +1,48 @@
 const BrowserUtils = require('./lib/BrowserUtils');
 const NativeUtils = require('./lib/NativeUtils');
-const wdioUI5 = require("./lib/wdioUi5-index");
-const logger = require("./lib/Logger");
-const selectorHelper = require("./lib/SelectorHelper");
+const wdioUI5 = require('./lib/wdioUi5-index');
+const cordovaPluginFactory = require('./lib/cordova-plugins/factory');
```
Comment (vobu) [inline index.js:4]: for better context + understanding, probably rename to "cordovaMockPluginFactory" or similar?

File: lib/cordova-plugins/factory.js (hunk 4 above)
Comment (dominikfeininger) [reply lib/cordova-plugins/factory.js:35]:
``` 
/**
     * call to trigger the init process
     * @param {index} _ WDI5 framework index.js
     * @param {WebdriverIO.BrowserObject} context
     */

File: test/wdio-shared.conf.js (hunk 8 above)
Comment (dominikfeininger) [reply test/wdio-shared.conf.js:105]: wdio-shared.conf.js is for native tests eg. ios, android... the wdio.conf.js is for browser testing.

File: index.js (hunk 10 above)
Comment (dominikfeininger) [reply index.js:4]: - renamed folder "cordova-plugins" to "cordova-plugin-mocks"
- changed variable name

**Pull request merged**